### ISSUE TITLE
Get GroupPosition Base on Its Child ChildPosition

### ISSUE THREAD

**Issue by Kuoyhout** [h6ah4i/android-advancedrecyclerview#270]
Hi! How do i get the child's group groupposition in onCreateChildViewHolder method?

**Comment by OzTK** [comment]
Position can't be accessed at this stage. Are you sure you can't do this operation in `onBindChildViewHolder()` ?

**Comment by Kuoyhout** [comment]
@OzTK I can't, because it causing error with log saying
`java.lang.IllegalStateException:Cannot call this method while RecyclerView is computing a layout or scrolling`

**Comment by OzTK** [comment]
What are you trying to do that causes this exception? In `onCreateChildViewHolder()` you are supposed to create an instance of your ViewHolder without having it linked to your data. Then, in `onBindChildViewHolder()`, you bind your item data to the ViewHolder passed. you also have direct access to groupPosition and childPosition in this method.

**Comment by Kuoyhout** [comment]
@OzTK I want to move the group and its child or the child in the group to the bottom position of the list, when the check box in the item view is check. But when I check the checkbox and scroll the recyclerview it cause that exception. 

So I try to do it like this for the groupViewHolder which fix the crash:

```
@Override
    public void onBindGroupViewHolder(MyGroupViewHolder holder, int groupPosition, int viewType) {
        onBindGroup = true;
        // group item
        ExampleGroupData item = (ExampleGroupData) mProvider.getGroupItem(groupPosition);
        // set listeners
        holder.itemView.setOnClickListener(mItemViewOnClickListener);

        // set text
        holder.mTextView.setText(item.getText());
        holder.checkBox.setChecked(item.isCheck());
        onBindGroup = false;
    }
```

```
public class MyGroupViewHolder extends MyBaseViewHolder {
        public MyGroupViewHolder(View v) {
            super(v);
            checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    if (isChecked && !onBindGroup) {
                        int thisGroupPosition = RecyclerViewExpandableItemManager.getPackedPositionGroup(mExpandableItemManager.getExpandablePosition(getAdapterPosition()));
                        // group item
                        final ExampleGroupData item = (ExampleGroupData) mProvider.getGroupItem(thisGroupPosition);
                        item.setCheck(true);
                        mProvider.updateGroupData(thisGroupPosition, item);
                        moveGroup(thisGroupPosition, getGroupCount() - 1);
                    }
                }
            });
        }
    }
```

But I don't know how to do the same for childViewHolder because I don't know how to get its groupPosition.

**Comment by OzTK** [comment]
RecyclerViewExpandableItemManager also allows you to get the group position for a child. Like this:
`RecyclerViewExpandableItemManager.getPackedPositionGroup(mExpandableItemManager. getExpandablePosition(getAdapterPosition))`

Is that what you need?

**Comment by h6ah4i** [comment]
@Kuoyhout Hi. I have just pushed a demo code to the [`demo_issue_270`](https://github.com/h6ah4i/android-advancedrecyclerview/tree/demo_issue_270) branch. 

I tweaked the "Expandable (Minimal)" example. Is it what you need? Please check the branch for more details. Thanks 😉 

![issue270demo](https://cloud.githubusercontent.com/assets/2552365/16357944/e9943bfc-3b3f-11e6-9264-770ceb1ac13d.gif)

**Comment by h6ah4i** [comment]
**[UPDATE]** @Kuoyhout I have added these three methods to `RecyclerViewExpandableItemManager` make move items operation easier.

- `notifyGroupItemMoved(int fromGroupPosition, int toGroupPosition)`
- `notifyChildItemMoved(int groupPosition, int fromChildPosition, int toChildPosition)`
- `notifyChildItemMoved(int fromGroupPosition, int fromChildPosition, int toGroupPosition, int toChildPosition)`


Could you try the latest develop branch code which includes the three methods? It is available via JitPack. Thanks.

```gradle
repositories {
    maven { url "https://jitpack.io" }
}

dependencies {
    compile 'com.github.h6ah4i:android-advancedrecyclerview:1b2ec40920b444e1903e14b5467923aa83dfd7cf'
}
```

**Comment by Kuoyhout** [comment]
@h6ah4i I've tested it! It works perfectly both the Group and Child item moved! 

But I got a warning on `viewholder.getLayoutPosition()` saying `Method invocation 'viewholder.getLayoutPosition()' may produce 'java.lang.NullPointerException'`

**Comment by h6ah4i** [comment]
@Kuoyhout Simply add a `null` check for the returned value of `findContainingItemView()`. This method may returns `null`.

```java
    @Nullable
    public View findContainingItemView(View view) {
        ViewParent parent = view.getParent();
        while (parent != null && parent != this && parent instanceof View) {
            view = (View) parent;
            parent = view.getParent();
        }
        return parent == this ? view : null;
    }
```

**Comment by Kuoyhout** [comment]
@OzTK Yes, That's what I'm looking for. 😄

**Comment by Kuoyhout** [comment]
@h6ah4i Thanks! Why don't you use `vh.getAdapterPosition()`? And When will this new version release?

**Comment by h6ah4i** [comment]
@Kuoyhout 

> Why don't you use vh.getAdapterPosition()? 
Oh yes, `vh.getAdapterPosition()` should be better in this situation.

> And When will this new version release?
I have no idea about this for now. On the next weekend, maybe.

**Comment by Kuoyhout** [comment]
@h6ah4i Ok Thanks again! 😃

**Comment by h6ah4i** [comment]
@Kuoyhout Hi. I have just merged the develop branch and released it as v0.9.2. It includes the new `notifyGroupItemMoved()` and `notifyChildItemMoved()` method finally! Thanks 😄

**Comment by Kuoyhout** [comment]
@h6ah4i Cool! Thank! 👍
